Assets/Stack mini game/Scripts/CubeSpawner.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CubeSpawner : MonoBehaviour
+{
+    [SerializeField] private GameObject cubePrefab;
+
+    [SerializeField] private Transform cubeSpawnStartPosition;
+
+    [SerializeField] private CubeStack cubeStack;
+
+    public Action<GameObject> CubeSpawnedEvent;
+
+    public void SpawnBottomCube()
+    {
+        GameObject spawnedCube = Instantiate(cubePrefab, cubeSpawnStartPosition.position, Quaternion.identity);
+
+        spawnedCube.transform.localScale = new Vector3(cubePrefab.transform.localScale.x, spawnedCube.transform.localScale.y, spawnedCube.transform.localScale.z);
+
+        CubeSpawnedEvent.Invoke(spawnedCube);
+    }
+
+    public void SpawnStackedCube(Vector3 spawnPosition, float setWidth)
```

Comment:
You could potentially combine these two spawn (bottom and stacked) methods into one as they do very similar things. Could pass optional parameters for the width and vector spawn offset and pass in different stuff for the bottom one versus the stacked one? There is some duplication here but it’s not too bad.

Reply:
Yeah I noticed. I think the idea here is that I wanted to keep the cubeSpawnStartPosition variable within the CubeSpawner class, versus adding it to the Gameplay manager. It felt more right this way for some reason, as the Gameplay manager uses the higher level components, not a specific position in the game. But again, it is more based on my personal impression than anything else.